hvplot/converter.py

```diff
@@ -559,9 +559,13 @@ def __init__(
             try:
                 if not use_dask:
                     symmetric = self._process_symmetric(symmetric, clim, check_symmetric_max)
-
                 if self._style_opts.get('cmap') is None:
-                    if symmetric:
+                    # Default to categorical camp if we detect categorical shading
+                    if (self.datashade and (self.aggregator is None or 'count_cat' in str(self.aggregator)) and
+                        ((self.by and not self.subplots) or
+                         (isinstance(self.y, list) or (self.y is None and len(self.variables) > 1)))):
```

Comment:
This test fails because the clause `self.y is None and len(self.variables) > 1` happens to be true. `self.variables` is equal to `['index, '0']`. I don't think this test should produce a categorical plot.

https://github.com/holoviz/hvplot/blob/ce0f4e135e0bb0c940a5092edb6d9436377b9c70/hvplot/tests/testoperations.py#L126-L129

Reply:
I'm confused; why is `self.y=None`? Shouldn't the y axis be the '0' column?

Reply:
The part of the code that is modified here in the `__init__` while for the case represented in the test `y` is only computed at a later stage when `__call__` is executed, in this specific method:

https://github.com/holoviz/hvplot/blob/ce0f4e135e0bb0c940a5092edb6d9436377b9c70/hvplot/converter.py#L1526

Reply:
I guess it should `len(self.variables) > 2` in that case.

Reply:
Or better yet `len(set(self.variables) - set(self.indexes)) > 1`